discord-bot/src/index.ts

```diff
@@ -0,0 +1,202 @@
+// // For Manifold SDK
+// process.env.NODE_ENV = 'production'
+
+import { REST } from '@discordjs/rest'
+import * as console from 'console'
+import {
+  ChatInputCommandInteraction,
+  Client,
+  Collection,
+  EmbedBuilder,
+  Events,
+  GatewayIntentBits,
+  MessageReaction,
+  PartialMessageReaction,
+  Partials,
+  PartialUser,
+  Routes,
+  SlashCommandBuilder,
+  TextChannel,
+  User,
+} from 'discord.js'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as process from 'process'
+import { fileURLToPath } from 'url'
+import { config } from './constants/config.js'
+import { getAnyHandledEmojiKey } from './emojis.js'
+import {
+  getMarketFromSlug,
+  getSlug,
+  handleReaction,
+  sendThreadErrorMessage,
+} from './helpers.js'
+import { registerApiKey } from './register-api-key.js'
+import { startListener } from './server.js'
+import { messagesHandledViaInteraction, userApiKey } from './storage.js'
+
+const { id: clientId } = config.client
+const token = process.env.DISCORD_BOT_TOKEN
+if (!token) throw new Error('No DISCORD_BOT_TOKEN env var set.')
+export const commands = new Collection<
+  string,
+  {
+    data: SlashCommandBuilder
+    execute: (interaction: ChatInputCommandInteraction) => Promise<any>
+  }
+>()
+const readCommandsFromFiles = async () => {
+  // Load commands
+  console.log('Loading commands... ')
+  const commandsPath = path.join(
+    path.dirname(fileURLToPath(import.meta.url)),
+    'commands'
+  )
+  const extension = commandsPath.includes('lib') ? '.js' : '.ts'
+  console.log('Commands path: ', commandsPath)
+  const commandFiles = fs
+    .readdirSync(commandsPath)
+    .filter((file: string) => file.endsWith(extension))
```

Comment:
I think this whole thing where it automatically enumerates all the command files and imports them dynamically is just worse than if you didn't do that. At least you are breaking typechecking, etc. on the interface for commands.

Reply:
Yup, weird way of doing things! Not sure why the documentation does that.

Reply:
Changed this to import commands and export them as an array from commands/index